open_city_profile/settings.py

```diff
@@ -337,6 +337,19 @@ def _list_to_string(value):
 AUDIT_LOG_FILENAME = env("AUDIT_LOG_FILENAME")
 
 if AUDIT_LOG_FILENAME:
+    if "X" in AUDIT_LOG_FILENAME:
+        import random
+        import string
+
+        system_random = random.SystemRandom()
+        char_pool = string.ascii_lowercase + string.digits
+        replaced = ""
+        for char in AUDIT_LOG_FILENAME:
+            if char == "X":
+                char = system_random.choice(char_pool)
+            replaced += char
+        AUDIT_LOG_FILENAME = replaced
```

Comment:
I know this is a personal preference, but manual per-character iteration feels a bit off  to me. I would replace this with something like

```python
AUDIT_LOG_FILENAME = re.sub('X', lambda x: system_random.choice(char_pool), AUDIT_LOG_FILENAME)
```

Reply:
I considered that but discarded it because using regexp for searching a single character feels a bit overkill. But I definitely would like a shorter code for this. I wish there was something like this available:

```python
AUDIT_LOG_FILENAME = replace_all('X', lambda x: system_random.choice(char_pool), AUDIT_LOG_FILENAME)
```

that wouldn't use regexps but just single character comparison. But alas I couldn't find such a thing from the standard library.

But then again, this code is executed in the startup of the program, so there's probably no worrying about using regexp for this either. I'll shorten this code, one way or another.